babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1583,3 +1583,49 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
             SqlLiteral.createBoolean(withTies, pos));
     }
 }
+
+SqlNode SqlHexCharStringLiteral() :
+{
+    final NlsString p;
+    final String formatString;
+    final String charSet;
+    CharacterSet charSetEnum = null;
```

Comment:
nit: is there a reason to make charSetEnum non-final and initialize it to null (nothing else is initialized here)? It looks like in the logic below, either this enum is set to a value, or an error is thrown, so it seems like there is no need for this initialization, but I could be missing something.

Reply:
There is an alternative path of matching <QUOTED_HEX_STRING> without ever assigning charSetEnum a value. The error only got throw when <PREFIXED_HEX_STRING_LITERAL> is match and the characterSet is not any one of those four.